Cargo.toml

```diff
@@ -22,20 +22,20 @@ rust.missing_debug_implementations = "warn"
 rust.unreachable_pub = "warn"
 rustdoc.all = "warn"
 rust.unused_must_use = "deny"
-rust.rust_2018_idioms = "deny"
+rust.rust_2018_idioms = { level = "deny", priority = -1 }
 
 [lints.clippy]
 # all lints that are on by default (correctness, suspicious, style, complexity, perf)
-all = "warn"
+all = { level = "warn", priority = -1 }
 
 # new lints that are still under development
-nursery = "warn"
+nursery = { level = "warn", priority = -1 }
 # avoid lints that are too pedantic
 future_not_send = "allow"
 fallible_impl_from = "allow"
 
 # lints which are rather strict or have occasional false positives
-pedantic = "warn"
+pedantic = { level = "warn", priority = -1 }
```

Comment:
Is there a reason to set the priority of these lints higher?

Reply:
It is something nice that was detected by the clippy on the latest version of Rust:

https://rust-lang.github.io/rust-clippy/master/#/lint_groups_priority

As we want to allow certain clippy, for overrides properly, we must set the clippy warning of the same category above with a lower priority. Quite well described in the error:

```shell
error: lint group `pedantic` has the same priority (0) as a lint
 --> Cargo.toml:38:1
 |
38 | pedantic = "warn"
 | ^^^^^^^^ ------ has an implicit priority of 0
39 | # avoid lints that are too pedantic
40 | must_use_candidate = "allow"
 | ------------------ has the same priority as this lint
 |
 = note: the order of the lints in the table is ignored by Cargo
 = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#lint_groups_priority
 = note: `-D clippy::lint-groups-priority` implied by `-D warnings`
 = help: to override `-D warnings` add `#[allow(clippy::lint_groups_priority)]`
help: to have lints override the group set `pedantic` to a lower priority
 |
38 | pedantic = { level = "warn", priority = -1 }
 | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: could not compile `kakarot-rpc` (lib) due to 1 previous error
```